specs/deployment.md

```diff
@@ -1,599 +1,166 @@
+## Environment Variables
 
-### Check KV Contents
+Non-sensitive configuration MAY be stored as environment variables in wrangler.toml under `[vars]` sections. These are public values that change between environments but are not secret.
 
-```bash
-# List OAuth tokens
-wrangler kv:key list --binding OAUTH_KV
-
-# Check rate limits
-wrangler kv:key list --binding RATE_LIMIT_KV
-```
-
----
+**GITHUB_ALLOWED_USER_ID** contains the GitHub user ID permitted to access the MCP server. This acts as a simple authorization allowlist and MUST be configured for both environments. Future versions MAY support multiple user IDs or different authorization mechanisms.
 
-## Rollback Procedures
+## Deployment Verification
 
-### Automated Rollback via GitHub Actions
+After any deployment, automated verification MUST confirm the Worker is accessible and responsive before considering the deployment successful.
 
-**Setup:** Create `.github/workflows/rollback.yml`:
+### Health Check Endpoint
 
-```yaml
-name: Rollback Deployment
+The Worker MUST expose a health check endpoint at `/health` that returns HTTP 200 with a JSON response indicating service status. This endpoint MUST NOT require authentication and SHOULD complete in under 100ms. The health check MUST verify:
 
-on:
-  workflow_dispatch:
-    inputs:
-      version:
-        description: 'Deployment version to rollback to'
-        required: true
-        type: string
+- Worker is running and responding to requests
+- All bindings (R2, KV, DO, Analytics) are accessible
+- No critical configuration errors prevent request handling
 
-jobs:
-  rollback:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          ref: ${{ inputs.version }}
+The health check MAY return degraded status (HTTP 200 with warnings) if non-critical features are unavailable (e.g., backup credentials missing but MCP tools functional).
 
-      - uses: actions/setup-node@v4
-        with:
-          node-version: '20'
+### Deployment Rollback
 
-      - run: pnpm install --frozen-lockfile
+The system MUST support rollback to a previous deployment within 10 minutes of detecting a failed deployment. Cloudflare Workers retains the last 10 deployments in history, allowing instant rollback via dashboard or API.
 
-      - name: Run tests
-        run: pnpm test
+Rollback MUST NOT require rebuilding or retesting the previous version. The Cloudflare Workers platform provides atomic deployment rollback that reverts to the exact previously deployed code.
 
-      - name: Deploy to production
-        run: npx wrangler deploy
-        env:
-          CLOUDFLARE_API_TOKEN: ${{ secrets.CLOUDFLARE_API_TOKEN }}
+Rollback MAY be triggered manually via Cloudflare dashboard or automatically via GitHub Actions if post-deployment verification fails. Automated rollback SHOULD only occur for critical failures (health check failure, authentication broken) not minor issues (increased error rates, performance degradation).
```

Comment:
We should never trigger the rollback from the Cloudflare dashboard. We should always do it through a workflow in GitHub as there are other actions we need to do at the same time.